compiler/src/iree/compiler/Dialect/Stream/Analysis/Partitioning/ReferencePartitioning.cpp

```diff
@@ -204,13 +254,13 @@ partitionStreamableOpsReference(IREE::Stream::PartitioningConfigAttr config,
     if (firstCandidateOrdinal != -1) {
       LLVM_DEBUG(llvm::dbgs() << "Moving to first candidate partition "
                               << firstCandidateOrdinal << " (continue)\n");
-      builders[firstCandidateOrdinal]->insert(&op);
-      opInfo.membership.set(firstCandidateOrdinal);
-      opInfo.hazards.reset(firstCandidateOrdinal);
+      builders[firstCandidateOrdinal]->insert(&op, opInfo);
       continue;
     }
 
     // Mark the op as having hazards against all other partitions.
+    // Why are we that conservative?
+    // Why we don't take the hazards for the users?
     if (!builders.empty()) {
       opInfo.hazards.set(0, builders.size() - 1);
     }
```

Comment:
@benvanik, could you shed some light on what was your reasoning here when you wrote the algorithm?

Reply:
It's better to be safe than incorrect, especially with our current minimal test coverage. It's not always safe to reorder things - if anything we are unlikely to be conservative enough here - for example, if there's a `stream.resource.load` of a resource or a global we can't move anything that may affect that resource or global. This partitioning was designed to be conservative because debugging such issues is _really_ difficult.